tests/Metadata/SnapshotMetadataTest.php

```diff
@@ -50,15 +51,40 @@ public function providerValidField() : array
     }
 
     /**
-     * {@inheritdoc}
+     * Tests that unsupported fields throw an exception.
+     *
+     * @param array $unsupportedField
+     *   The array of nested keys for the unsupported field.
+     * @param mixed $fieldValue
+     *   The field value to set.
+     *
+     * @dataProvider providerUnsupportedFields
+     *
+     * @return void
+     */
+    public function testUnsupportedFields(array $unsupportedField, $fieldValue):void
+    {
+        $metadata = json_decode($this->localRepo[$this->validJson], true);
+        $this->nestedChange($unsupportedField, $metadata, $fieldValue);
+        $fieldName = array_pop($unsupportedField);
+        $expectedMessage = preg_quote("Object(ArrayObject)[signed][meta][targets.json][$fieldName]", '/');
+        $expectedMessage .= ".*This field is not supported.";
+        $this->expectException(MetadataException::class);
+        $this->expectExceptionMessageMatches("/$expectedMessage/s");
+        static::callCreateFromJson(json_encode($metadata));
```

Comment:
It'd be nice if MetadataBase had a `createFromArray()` method, so that we can pass it stuff like this without needing to re-encode it as JSON. Not a big deal, though!

Reply:
That would need to be public method which I don't think we should add just for testing purposes. But maybe we could add `createMetadataFromArray()` to `\Tuf\Tests\Metadata\MetaDataBaseTest` though in another PR